commit fc5a7f0339d0641c0df14a9ed8af49c9992f9bef
Author: Gustavo A. R. Silva <gustavo@embeddedor.com>
Date:   Fri Mar 9 12:27:08 2018 -0600

    power: supply: da9150-fg: remove VLA usage
    
    In preparation to enabling -Wvla, remove VLA usage and replace it
    with fixed-length arrays.
    
    DA9150_QIF_LONG_SIZE (4 bytes) is the biggest size of an attribute which can
    be accessed [1].
    
    Fixed as part of the directive to remove all VLAs from
    the kernel: https://lkml.org/lkml/2018/3/7/621
    
    [1] https://marc.info/?l=kernel-hardening&m=152059600524753&w=2
    
    Signed-off-by: Gustavo A. R. Silva <gustavo@embeddedor.com>
    Acked-by: Adam Thomson <Adam.Thomson.Opensource@diasemi.com>
    Signed-off-by: Sebastian Reichel <sebastian.reichel@collabora.co.uk>

diff --git a/drivers/power/supply/da9150-fg.c b/drivers/power/supply/da9150-fg.c
index 8b8ce978656a..1e2e5b0520c9 100644
--- a/drivers/power/supply/da9150-fg.c
+++ b/drivers/power/supply/da9150-fg.c
@@ -92,7 +92,7 @@ struct da9150_fg {
 static u32 da9150_fg_read_attr(struct da9150_fg *fg, u8 code, u8 size)
 
 {
-	u8 buf[size];
+	u8 buf[DA9150_QIF_LONG_SIZE];
 	u8 read_addr;
 	u32 res = 0;
 	int i;
@@ -111,7 +111,7 @@ static void da9150_fg_write_attr(struct da9150_fg *fg, u8 code, u8 size,
 				 u32 val)
 
 {
-	u8 buf[size];
+	u8 buf[DA9150_QIF_LONG_SIZE];
 	u8 write_addr;
 	int i;